cg/services/delivery_message/utils.py

```diff
@@ -87,3 +89,11 @@ def validate_cases(cases: list[Case], case_ids: list[str]) -> None:
 
 def is_matching_order(cases: list[Case]) -> bool:
     return all([case.latest_order == cases[0].latest_order for case in cases])
+
+
+def get_cases_ready_for_delivery(trailblazer_api: TrailblazerAPI, order: Order) -> list[Case]:
+    return [
+        case
+        for case in order.cases
+        if trailblazer_api.get_latest_analysis_status(case.id) == AnalysisStatus.COMPLETED
```

Comment:
Do we want to query tb with order_id and status instead? Would require a new method in the TrailblazerAPI

Reply:
I like that idea, sounds cleaner and more efficient! But stick to this for now if you want

Reply:
Can a case be part of different orders? If so, not specifying the order could perhaps fetch something unwanted. That being said, I am not up to speed with the order logic.

Reply:
Yeah, I guess we could have that unlucky scenario @ChrOertlin. Seems as if everything points to some refactoring!

Reply:
Refactored